File: pset6/caesar/caesar.py
import sys
from sys import argv
import cs50


def main():

    # Checking for main argc to be 2
    if len(argv) == 2:
        # Checking over the input chars.
        c_flag = check(len(sys.argv[1]), sys.argv[1])

        # Sequence for valid input only
        if c_flag == True:
            # Shift amount transformed from String to int
            typed = sys.argv[1]
            key = []

            Dec = 0
            summation = 0
            n_shift = len(typed)
            multi = 0
            times = n_shift - 1

            for i in range(n_shift):

                multi = 0
                Dec = int(typed[i]) % 48

                if multi == times:
                    Dec = Dec * 1

                else:
                    while multi < times:

                        Dec = Dec * 10
                        multi += 1

                key.append(Dec)
                summation = summation + Dec
                times -= 1

            # Cipher Generation
            plain = cs50.get_string("plain text: ")
            s = []

            # Ciphertext generating
            for i in plain:

                if i >= 'A' and i <= 'Z':

                    if ord(i) + summation > 90:

                        while ord(i) + summation > 90:
                            i = chr((ord(i) + summation) % 91 + 65)
                            if i >= 'A' and i <= 'Z':
                                break
                            else:
                                i = chr(ord(i) % 123 + 97)

                        s.append(i)

                    else:
                        i = chr((ord(i) + summation) % 91)
                        s.append(i)

                # Small letters
                elif i >= 'a' and i <= 'z':

                    if (ord(i) + summation) > 122:

                        i = chr((ord(i) + summation) % 123 + 97)

                        while True:
                            if i >= 'a' and i <= 'z':
                                break
                            else:
                                i = chr(ord(i) % 123 + 97)

                        s.append(i)

                    else:
                        i = chr((ord(i) + summation) % 123)
                        s.append(i)

                else:
                    s.append(i)

            c = ''.join(s)
            print(f"ciphertext: {c}")

        else:
            print(" Usage: ./caesar key ")
            sys.exit(1)

    else:
        print(" Usage: ./caesar key ")
        sys.exit(1)


# Check for validation of input
def check(length, nano):

    check = 0
    for i in range(length):
        c = nano[i]

        # Checking for inputs rather than numbers
        if c >= '0' and c <= '9':
            continue
        else:

            i = length
            check = 1

    if check == 1:

        return False

    else:

        return True


if __name__ == "__main__":
    main()

File: Pset2/Caesar encyption/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
bool check(int length, string nano[]);
int main(int argc, string argv[])
{
    bool c_flag ;
    char x;
    // Checking for main argc to be 2
    if (argc == 2)
    {
        // Checking over the input chars.
        c_flag = check(strlen(argv[1]), &argv[1]);
        
        // Sequence for valid input only 
        if (c_flag == true)
        {   
            // Shift amount transformed from String to int
            
            printf("%s\n",argv[1]);
            string typed =argv[1];
            int key[strlen(typed)];

            int Dec = 0, sum = 0,  n_shift = strlen(typed) , multi =0 , times = n_shift-1;

            for(int i = 0 ; i < n_shift ; i++)
            {
                multi =0;
                Dec= typed[i] % 48;
                printf("the character processed is %i \n",Dec);
                if(multi == times)
                {
                    Dec = Dec*1;
                    printf("the summation is %i \n",Dec);
                }
                else
                {
                    while( multi < times)
                    {
                        printf("m is %i \n",multi);
                        Dec = Dec*10;
                        printf("the summation is %i \n",Dec);
                        multi++;
                    }
                }
                key[i] = Dec;  
                sum= sum + Dec;
                times--;
            }
               printf("the shift  is %i \n",sum);

            // Cipher Generation
            string plain  = get_string("plain text: \t");     
            string s= plain;
            //show the string again
            for(int i = 0, n = strlen(plain) ; i < n ; i++)
            {
                printf("%c\t",plain[i]);
                //printf("the number BECOMES  %c \n",plain[i]+1);
                if((int)plain[i] >= 65 && (int)plain[i] <= 90)
                {
                    if(((int)plain[i]+sum ) > 90)
                    {
                    x= (plain[i]+sum )%91 +65;
                   // x=(char)(c);
                   // printf("the character saved is %c \n",x);
                    s[i] = x;       
                    }
                    else
                    {
                        x= (plain[i]+sum )%91;
                       // x=(char)(c);
                        //printf("the character saved is %c \n",x);
                        s[i] = x;  
                    }
                }
                // Small letters
                else if((int)plain[i] >= 97 && (int)plain[i] <= 122)
                {
                    if(((int)plain[i]+sum ) > 122)
                    {
                        x= (plain[i]+sum )%123 +97;
                        //x=(char)(c);
                        //printf("the character saved is %c \n",x);
                        s[i] = x;       
                    }
                    else
                    {
                        x= ((int)plain[i]+sum )%123;
                        //x=(char)(c);
                        //printf("the character saved is %c \n",x);
                        s[i] = x;  
                    }    
                }
                else
                {
                }
            }
            printf("\n");
            printf("the whole cipher \" %s \"  \n",s);          
                }
        else
        {
            printf(" Usage: ./caesar key \n");
        }
        
    }
    else 
    {
        printf(" Usage: ./caesar key \n");
    }
}



// Check for validation of input
bool check(int length, string nano[])
{
    // Printf(" string = %s \n", nano[0]);
    int c, check = 0 ;
    for (int i = 0 ; i < length ; i++)
    {
        c = (int) nano[0][i];
        //checking for inputs rather than numbers
        if (c >= 48 && c <= 57)
        {
            //    printf("valid input \n");
        }
        else
        {                
            i = length;
            // Printf("INVALID INPUT \n");
            check = 1;
        }
    }
    if (check == 1)
    {
        return false;
    }
    else
    {
        return true;
    }
}
